src/helpers/purchase-result-helpers.ts

```diff
@@ -0,0 +1,25 @@
+import type { CustomerInfo } from "../entities/customer-info";
+import type { Package } from "../entities/offerings";
+
+export function getRenewalDateFromPackage(
+  customerInfo: CustomerInfo,
+  rcPackage: Package,
+): Date | null {
+  const productIdentifier = rcPackage.webBillingProduct.identifier;
+
+  const isActiveSubscription =
+    customerInfo.activeSubscriptions.has(productIdentifier);
+
+  if (!isActiveSubscription) {
```

Comment:
would be cool to actually refactor `getWillRenew` to ensure that this is a purchase that renews although for this use case the function will be called right after purchase so not sure if worth it, thoughts @tonidero ?

Reply:
Hmm not sure I understand what you mean... `getWillRenew` already checks for the expiration date but that of an entitlement, instead of a product directly: https://github.com/RevenueCat/purchases-js/blob/bd079f5ac814e86bf163b35373cc3c99d05ee8f5/src/entities/customer-info.ts#L325 . Note that there can be uncommon situations where more than one subscription gives the same entitlement, since it's an N to M relationship).

Entitlements are not recommended to be used with consumables, since the model doesn't really match though, so this is mostly only checked for subscriptions + non consumables.

Lmk if I misunderstood this though!